dashboard/lib/sample_data.rb

```diff
@@ -0,0 +1,199 @@
+  # and recreate. Sections and followers would be soft-deleted by
+  # dependency when we delete the teacher; but to not leave a trail of
+  # old test data behind, we explictly hard-delete.
+  def self.create_teacher(email, password, name)
+    # Delete any existing test data
+    user = User.find_by_email_or_hashed_email(email)
+    unless user.nil?
+      user.sections.each do |section|
+        # Hard-delete all students in each section.
+        section.students.each do |student_user|
+          raise "Not a sample student" unless student_user.name =~ SAMPLE_STUDENT_NAME_REGEX
+          raise "Should not be run outside of development" unless CDO.rack_env?(:development)
+          UserGeo.where(user_id: student_user.id).destroy_all
+          student_user.really_destroy!
+        end
+        # Hard-delete each section.
+        section.really_destroy!
+      end
+      UserGeo.where(user_id: user.id).destroy_all
+      # Delete the existing test teacher
+      raise "Not a sample teacher" unless user.name.eql? SAMPLE_TEACHER_NAME
+      raise "Not a sample teacher" unless user.email.eql? SAMPLE_TEACHER_EMAIL
+      raise "Should not be run outside of development" unless CDO.rack_env?(:development)
+      user.really_destroy!
+    end
+    # Create the test teacher
+    create :teacher, email: email, name: name,
+      password: password, terms_of_service_version: 1
+  end
+
+  # Generate a random gender choice with reasonable distributions
+  def self.random_gender
+    val = rng.rand(100)
+    gender = nil
+    case val
+    when 0..29
+      gender = "m"
+    when 30..59
+      gender = "f"
+    when 60..89
+      gender = nil
+    when 90..94
+      gender = "n"
+    when 95..99
+      gender = "o"
+    end
+    gender
+  end
+
+  # Create a test section
+  # Options is a hash with expected values:
+  #  :teacher - User object of teacher
+  #  :name - section name
+  #  :login_type - Section::LOGIN_TYPE_xxx value of login type for section
+  #  :grade - grade for section
+  #  :age_min - minimum age to generate for students
+  #  :age_max_inclusive - max age (inclusive) to generate for students
+  #  :script_name - name of script to assign for section
+  #  :num_students - number of students to generate for section
+  #  :use_imperfect_results - if true, generate some less than perfect
+  #    results. (CSF allows imperfect results, CSD and CSP do not.)
+  def self.create_section(options)
+    script = Script.get_from_cache(options[:script_name])
+    level_count = script.script_levels.count
+
+    # Create the section
+    section = create :section, script: script,
+      **options.slice(:teacher, :name, :login_type, :grade)
+
+    current_student = 0
+
+    # Create students in section
+    (1..options[:num_students]).each do
+      # Choose random properties and create student
+      age_min = options[:age_min]
+      age_max_inclusive = options[:age_max_inclusive]
+      current_student += 1
```

Comment:
Note this will reset the count for each section, reusing the student names between sections. Names don't have to be unique so that's ok but it might be clearer / easier to distinguish if we increase the number for all students in the seed (across all sections).

Reply:
Thanks. I did this deliberately because it seems less confusing to look at sections this way (with numbering within sections). If we prefer it "globally numbered" we can make that change later.